lib/Queue/PushRedisFactory.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+declare(strict_types=1);
+/**
+ * SPDX-FileCopyrightText: 2025 Robin Appelman <robin@icewind.nl>
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+namespace OCA\NotifyPush\Queue;
+
+use OC\RedisFactory;
+use OCP\IConfig;
+
+class PushRedisFactory {
+	private $instance = null;
+	private IConfig $config;
+	private RedisFactory $redisFactory;
+
+	public function __construct(
+		IConfig $config,
+		RedisFactory $redisFactory,
+	) {
+		$this->config = $config;
+		$this->redisFactory = $redisFactory;
+	}
```

Comment:
```suggestion

	public function __construct(
		private readonly IConfig $config,
		private readonly RedisFactory $redisFactory,
	) {}
```

Reply:
I like the idea but `notify_push` is compatible with Nextcloud 27 which does not necessarily support PHP 8.1 (required for readonly properties).